Framework/MDAlgorithms/src/SaveMD2.cpp

```diff
@@ -133,10 +133,10 @@ void SaveMD2::doSaveHisto(Mantid::DataObjects::MDHistoWorkspace_sptr ws) {
   for (size_t d = 0; d < numDims; d++) {
     std::vector<double> axis;
     IMDDimension_const_sptr dim = ws->getDimension(d);
-    for (size_t n = 0; n < dim->getNBins() + 1; n++)
+    for (size_t n = 0; n < dim->getNBoundaries(); n++)
```

Comment:
Move `dim->getNBoundaries()` out of the for loop. Compilers have a hard time seeing through virtual functions.